src/miner_lora_throttle.erl

```diff
@@ -0,0 +1,241 @@
+%% @doc This module provides time-on-air regulatory compliance for the
+%% EU868 and US915 ISM bands.
+%%
+%% This module does not interface with hardware or provide any
+%% transmission capabilities itself. Instead, the API provides its
+%% core functionality through `track_sent/4', `can_send/4', and
+%% `time_on_air/6'.
+-module(miner_lora_throttle).
+
+-export([
+    can_send/4,
+    dwell_time/3,
+    dwell_time_period/1,
+    max_dwell_time/1,
+    new/1,
+    time_on_air/6,
+    track_sent/9,
+    track_sent/4
+]).
+
+-export_type([
+    region/0,
+    handle/0
+]).
+
+-record(sent_packet, {
+    sent_at :: number(),
+    time_on_air :: number(),
+    frequency :: number()
+}).
+
+-type region() :: 'EU868' | 'US915' | undefined.
+
+-opaque handle() :: {region(), list(#sent_packet{})}.
+
+%% @doc Time over which we keep sent packet statistics for duty-cycle
+%% limited regions (EU868).
+%%
+%% In order to calculate duty cycle, we track every single
+%% transmission 'now' and the previous DUTY_CYCLE_PERIOD_MS of
+%% time. Note that 'now' is always changing and that transmissions
+%% older than DUTY_CYCLE_PERIOD_MS are only untracked when updating
+%% state or calculating duty-cycle.
+-define(DUTY_CYCLE_PERIOD_MS, 3600000).
+
+%% Maximum time on air for dell-time limited regions (US915).
```

Comment:
dwell typo

Reply:
thanks. Emacs is supposed to catch these, but dell is word.